File: src/08.ts
const isSmallLatinAlphabet = (char: string) => {
    const smallACode = 'a'.charCodeAt(0)
    const smallZCode = 'z'.charCodeAt(0)
    const charCode = char.charCodeAt(0)
    return charCode >= smallACode && charCode <= smallZCode
}

const cipher08 = (base: number, source: string): string => {
    return source
        .split('')
        .map((char): string => {
            if (isSmallLatinAlphabet(char))
                return String.fromCharCode(base - char.charCodeAt(0))
            else
                return char
        })
        .join('')
}

{
    const ciphered = cipher08(219, 'The quick brown fox jumps over the lazy dog')
    console.log(ciphered)
    const deciphered = cipher08(219, ciphered)
    console.log(deciphered)
}

File: src/08.py
def cipher_08(base: int, source: str) -> str:
    result = ""
    for source_char in source:
        code = ord(source_char)
        if ord("a") <= code <= ord("z"):
            result += chr(base - code)
        else:
            result += source_char
    return result


ciphered = cipher_08(219, "The quick brown fox jumps over the lazy dog")
print(ciphered)
deciphered = cipher_08(219, ciphered)
print(deciphered)
